sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -10,11 +10,26 @@ import { AbortSignalLike } from '@azure/abort-controller';
 export type CancelOnProgress = () => void;
 
 // @public
-export interface LongRunningOperation<T> {
-    requestMethod: string;
-    requestPath: string;
-    sendInitialRequest: () => Promise<LroResponse<T>>;
-    sendPollRequest: (path: string) => Promise<LroResponse<T>>;
+export function createPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreatePollerOptions<TResult, TState>): Promise<SimplePollerLike<TState, TResult>>;
+
+// @public
+export interface CreatePollerOptions<TResult, TState> {
+    intervalInMs?: number;
+    processResult?: (result: unknown, state: TState) => TResult;
+    resourceLocationConfig?: LroResourceLocationConfig;
+    restoreFrom?: string;
```

Comment:
do you think `restoreFrom` is useful here? I was thinking we could take it out of `CreatePollerOptions` and have a function exported for restoring `restorePoller(state: string)`. What do you think?

Reply:
From a customer point of view, I agree they should be able to restore a poller with just the serialized state. However, from an implementation point of view, the serialized state is not enough to restore a poller, for instance, the implementation still needs to know how to update the state after every poll and how to transform the final result. For context, here is how `createPoller` is used to implement the `restoreAnalyzebatchPoller` method:
```ts
    const poller = await createPoller(lro, {
      intervalInMs: updateIntervalInMs,
      restoreFrom: serializedState,
      processResult: processAnalyzeResult({
        client: this._client,
        tracing: this._tracing,
        documents,
        opOptions: { ...rest, includeStatistics },
        state,
      }),
      updateState: createUpdateAnalyzeState(),
      withPollingUrl(currentPollingUrl) {
        state.continuationToken = currentPollingUrl;
      },
    });
```
In fact, it looks for similar to how `createPoller` is used in `beginAnalyzeBatch`:
```ts
    const poller = await createPoller(lro, {
      intervalInMs: updateIntervalInMs,
      processResult: processAnalyzeResult({
        client: this._client,
        tracing: this._tracing,
        documents: realInputs,
        opOptions: { ...rest, includeStatistics },
        state,
      }),
      updateState: createUpdateAnalyzeState(realInputs),
      withPollingUrl(currentPollingUrl) {
        state.continuationToken = currentPollingUrl;
      },
    });
```
The only main difference is in the former takes `restoreFrom` too as input.

Reply:
I agree with Deya. We need to know both the definition of the operation and the necessary elements of the state to restore it, but I think we may need even lower-level control over how the operation is serialized and restored. While we may be able to restore the poller in this way for all LroEngine patterns, in general it feels like `restore` is just a modified way to initialize the poller instance.

Reply:
I will mark this as resolved but let me know if you disagree.